frontend/packages/multicluster-sdk/src/components/FleetResourceLink.tsx

```diff
@@ -0,0 +1,253 @@
+  const isHubCluster = !cluster || cluster === hubClusterName
+  const isMulticloudPath = location.pathname.startsWith('/multicloud/')
+
+  // helper function for ManagedCluster routing
+  const getManagedClusterPath = (name: string): { path: string | null; shouldFallback: boolean } => {
+    const firstClassPath = `/multicloud/infrastructure/clusters/details/${name}/${name}/overview`
+
+    if (isHubCluster) {
+      if (isMulticloudPath) {
+        return { path: firstClassPath, shouldFallback: false }
+      }
+      return { path: null, shouldFallback: true }
+    }
+    return { path: firstClassPath, shouldFallback: false }
+  }
+
+  // helper function for extension-based routing
+  const getExtensionPath = (
+    kind: string,
+    group: string | undefined,
+    version: string | undefined,
+    name: string
+  ): string | null => {
+    if (!resourceRoutesResolved) return null
+
+    const handler = getResourceRouteHandler(group, kind, version)
+    if (!handler || typeof handler !== 'function') return null
+
+    return handler({
+      kind,
+      cluster: cluster ?? hubClusterName,
+      namespace,
+      name,
+    })
+  }
+
+  // helper function for hub cluster context awareness
+  const shouldUseExtensionPath = (extensionPath: string | null): boolean => {
+    if (!extensionPath) return false
+    if (!isHubCluster) return true
+    return isMulticloudPath
+  }
+
+  // helper function for managed cluster search path
+  const getManagedClusterSearchPath = (): string => {
+    return `/multicloud/search/resources${getURLSearchParam({
+      cluster,
+      kind: groupVersionKind?.kind,
+      apigroup: groupVersionKind?.group,
+      apiversion: groupVersionKind?.version,
+      name,
+      namespace,
+    })}`
+  }
+
+  // shared helper function for extension-based resource routing
+  const getExtensionBasedResourcePath = (name: string): { path: string | null; shouldFallback: boolean } => {
+    const extensionPath = getExtensionPath(
+      groupVersionKind?.kind ?? '',
+      groupVersionKind?.group,
+      groupVersionKind?.version,
+      name
+    )
+
+    if (shouldUseExtensionPath(extensionPath)) {
+      return { path: extensionPath, shouldFallback: false }
+    }
+
+    if (extensionPath && isHubCluster && !isMulticloudPath) {
+      return { path: null, shouldFallback: true }
+    }
+
+    if (isHubCluster) {
+      return { path: null, shouldFallback: true }
+    }
+
+    return { path: getManagedClusterSearchPath(), shouldFallback: false }
+  }
+
+  // helper function for VirtualMachine routing
+  const getVirtualMachinePath = (name: string): { path: string | null; shouldFallback: boolean } => {
+    return getExtensionBasedResourcePath(name)
+  }
+
+  // helper function for generic resource routing
+  const getGenericResourcePath = (name: string): { path: string | null; shouldFallback: boolean } => {
+    return getExtensionBasedResourcePath(name)
+  }
```

Comment:
These wrapper functions do nothing - as I mentioned elsewhere, we shouldn't need any special handling for VMs.